src/Sarif.Viewer.VisualStudio/ProjectNameCache.cs

```diff
@@ -1,46 +1,41 @@
-﻿using System.Collections.Generic;
+﻿using EnvDTE;
+using System.Collections.Generic;
 
 namespace Microsoft.Sarif.Viewer
 {
     public sealed class ProjectNameCache
```

Comment:
>ProjectNameCache [](start = 24, length = 16)

- No longer a Singleton.
- Locking no longer needed.
- Solution injected into ctor to enable unit testing.
- `SetName` can now be private.

Reply:
Good changes.  Even moved the private methods to the bottom!

---
In reply to: [135426252](https://github.com/Microsoft/sarif-sdk/pull/599#discussion_r135426252) [](ancestors = 135426252)

Reply:
:-)

---
In reply to: [135559799](https://github.com/Microsoft/sarif-sdk/pull/599#discussion_r135559799) [](ancestors = 135559799,135426252)